baseplate/frameworks/thrift/__init__.py

```diff
@@ -51,13 +52,22 @@ def call_with_context(*args: Any, **kwargs: Any) -> Any:
                 span.finish(exc_info=sys.exc_info())
                 raise
             except Error as exc:
+                c = ErrorCode()
+                status = c._VALUES_TO_NAMES.get(exc.code, "")
+                span.set_tag("exception_type", "Error")
+                span.set_tag("thrift.status_code", exc.code)
+                span.set_tag("thrift.status", status)
+                span.set_tag("success", "false")
                 # mark 5xx errors as failures since those are still "unexpected"
-                if exc.code // 100 == 5:
+                if 500 <= exc.code < 600:
                     span.finish(exc_info=sys.exc_info())
                 else:
                     span.finish()
                 raise
             except TException:
+                exc_info = sys.exc_info()
```

Comment:
I think this could be 
```suggestion
   except TExecption as e:
   span.set_tag("exception_type", type(e).__name__)
```